packages/router-plugin/src/router.module.ts

```diff
@@ -4,15 +4,14 @@ import { NgxsModule } from '@ngxs/store';
 import { RouterState } from './router.state';
 import { DefaultRouterStateSerializer, RouterStateSerializer } from './serializer';
 
-@NgModule()
+@NgModule({
+  imports: [NgxsModule.forFeature([RouterState])]
+})
 export class NgxsRouterPluginModule {
-  static forRoot(): ModuleWithProviders[] {
-    return [
-      NgxsModule.forFeature([RouterState]),
-      {
-        ngModule: NgxsRouterPluginModule,
-        providers: [{ provide: RouterStateSerializer, useClass: DefaultRouterStateSerializer }]
-      }
-    ];
+  static forRoot(): ModuleWithProviders {
```

Comment:
> It is mandatory to use the generic ModuleWithProviders<T> type to indicate the Angular module type.
```suggestion
  static forRoot(): ModuleWithProviders<put type here> {
```

Reply:
I can't put type, because Angular 5 build failed

Reply:
This is required by Ivy, f*** this Angular 5 :joy: :joy:

Reply:
CI failed if added

Reply:
Hmmm... I'm feeling more and more like we need to drop support for it in this release.

Reply:
Let's drop NG5 support in a separate PR and then add this generic type there.
We should include the reasons for dropping support in that PR.